### PR TITLE
Benjamin sprint 2

### PR DESCRIPTION
**Pull request by Isildur512**
I added basic level transitions

### CODE REVIEW

File: Gaming the Syllabus/Assets/Scripts/Portal.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class Portal : MonoBehaviour
+{
+    public string levelName;
```
Comment (travis-bittel) [inline Gaming the Syllabus/Assets/Scripts/Portal.cs]: Standard practice is to keep all variables private for encapsulation and add the [SerializeField] attribute if it needs to be visible in the inspector

File: Gaming the Syllabus/Assets/Scripts/Portal.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
```
Comment (travis-bittel) [inline Gaming the Syllabus/Assets/Scripts/Portal.cs:5]: One thing you can consider is to use [RequireComponent(Collider2D)] to make it so this script can only be attached if the object has a collider. Minor thing, but can help explain what the component does and avoid some mistakes in usage

File: Gaming The Syllabus/Assets/Scripts/LevelLoader.cs [hunk 3]
```diff
@@ -9,6 +9,7 @@ public class LevelLoader : MonoBehaviour
 {
     public Animator transition;
     public string curLevel;
+    public Scene toLoad;
```
Comment (travis-bittel) [inline Gaming The Syllabus/Assets/Scripts/LevelLoader.cs]: Should be private with [SerializeField]

File: Gaming The Syllabus/Assets/Scripts/LevelLoader.cs [hunk 4]
```diff
@@ -19,34 +20,33 @@ void Update()
             LoadNextLevel();
         }
         */
-        if(GameObject.Find("Portal").transform.position.y <=GameObject.Find("Player").transform.position.y ){
-            LoadNextLevel();
-        
-            
-            
+        if(transform.position == GameObject.Find("Player").transform.position) {
```
Comment (travis-bittel) [inline Gaming The Syllabus/Assets/Scripts/LevelLoader.cs]: GameObject.Find("Player") in Update is extremely intensive. Let's try calling this once in Start and storing the result in a variable to use later.

**Review by travis-bittel — CHANGES_REQUESTED** [review]

File: Gaming the Syllabus/Assets/Scripts/Portal.cs (hunk 1 above)
Comment (rjmm99) [reply Gaming the Syllabus/Assets/Scripts/Portal.cs]: quick question, so "[SerializeField]" simply let you change/set values from the inspector?

Comment (travis-bittel) [reply Gaming the Syllabus/Assets/Scripts/Portal.cs]: Technically it forces Unity to serialize a private field. I'm unfamiliar with the specifics, but it basically just makes a private field show up in the inspector. Public fields show up there by default

File: Gaming The Syllabus/Assets/Scripts/LevelLoader.cs (hunk 4 above)
Comment (Isildur512) [reply Gaming The Syllabus/Assets/Scripts/LevelLoader.cs]: Does it not need to be checked constantly?

Comment (travis-bittel) [reply Gaming The Syllabus/Assets/Scripts/LevelLoader.cs]: Once you have a reference to the player GameObject you can use player.transform.position which will be the current position

Comment (travis-bittel) [reply Gaming The Syllabus/Assets/Scripts/LevelLoader.cs]: In Start:
`GameObject player = GameObject.Find("Player");`

Everywhere else:
`if (transform.position == player.transform.position) {
    . . .
}`

**Pull request merged**